File: projectBlueWater/src/main/java/com/lasthopesoftware/bluewater/client/stored/library/specs/GivenASetOfLibraries/AndSomeSyncOnlyOnLocalConnections/WhenGettingAllLibraries.java
package com.lasthopesoftware.bluewater.client.stored.library.specs.GivenASetOfLibraries.AndSomeSyncOnlyOnLocalConnections;

import com.lasthopesoftware.bluewater.client.library.access.specs.FakeLibraryProvider;
import com.lasthopesoftware.bluewater.client.library.repository.Library;
import com.lasthopesoftware.bluewater.client.stored.library.SyncLibraryProvider;
import com.lasthopesoftware.bluewater.shared.promises.extensions.specs.FuturePromise;

import org.junit.BeforeClass;
import org.junit.Test;

import java.util.Arrays;
import java.util.Collection;
import java.util.concurrent.ExecutionException;

import static org.assertj.core.api.Assertions.assertThat;

public class WhenGettingAllLibraries {

	private static final Collection<Library> expectedLibraries = Arrays.asList(
		new Library().setId(5).setIsSyncLocalConnectionsOnly(true).setLocalOnly(true),
		new Library().setId(4),
		new Library().setId(8),
		new Library().setId(99).setIsSyncLocalConnectionsOnly(true).setLocalOnly(true),
		new Library().setId(13));
	private static Collection<Library> libraries;

	@BeforeClass
	public static void context() throws ExecutionException, InterruptedException {
		final SyncLibraryProvider syncLibraryProvider = new SyncLibraryProvider(
			new FakeLibraryProvider(Arrays.asList(
				new Library().setId(5).setIsSyncLocalConnectionsOnly(true),
				new Library().setId(4),
				new Library().setId(8),
				new Library().setId(99).setIsSyncLocalConnectionsOnly(true),
				new Library().setId(13))));

		libraries = new FuturePromise<>(syncLibraryProvider.getAllLibraries()).get();
	}

	@Test
	public void thenTheLibraryIsNotLocalOnly() {
		assertThat(libraries).containsExactlyElementsOf(expectedLibraries);
	}
}

File: projectBlueWater/src/test/java/com/lasthopesoftware/bluewater/client/stored/library/GivenASetOfLibraries/AndSomeSyncOnlyOnLocalConnections/WhenGettingAllLibraries.kt
package com.lasthopesoftware.bluewater.client.stored.library.GivenASetOfLibraries.AndSomeSyncOnlyOnLocalConnections

import com.lasthopesoftware.bluewater.client.browsing.library.access.FakeLibraryRepository
import com.lasthopesoftware.bluewater.client.browsing.library.repository.Library
import com.lasthopesoftware.bluewater.client.stored.library.SyncLibraryProvider
import com.lasthopesoftware.bluewater.shared.promises.extensions.ExpiringFuturePromise
import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test

class WhenGettingAllLibraries {
	private val expectedLibraries = listOf(
		Library(id = 5, isSyncLocalConnectionsOnly = true, isLocalOnly = true),
		Library(id = 4),
		Library(id = 8),
		Library(id = 99, isSyncLocalConnectionsOnly = true, isLocalOnly = true),
		Library(id = 13)
	)

	private val libraries by lazy {
		val syncLibraryProvider = SyncLibraryProvider(
			FakeLibraryRepository(
				Library(id = 5, isSyncLocalConnectionsOnly = true),
				Library(id = 4),
				Library(id = 8),
				Library(id = 99, isSyncLocalConnectionsOnly = true),
				Library(id = 13)
			)
		)
		ExpiringFuturePromise(syncLibraryProvider.allLibraries).get()
	}

	@Test
	fun `then the libraries are correct`() {
		assertThat(libraries).containsExactlyElementsOf(expectedLibraries)
	}
}
